File: ios/App/App/PullToRefreshPlugin.swift
import Capacitor

@objc(PullToRefreshPlugin)
public class PullToRefreshPlugin: CAPInstancePlugin, CAPBridgedPlugin {
    public let identifier = "PullToRefreshPlugin"
    public let jsName = "PullToRefresh"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "beginRefresh", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "endRefresh", returnType: CAPPluginReturnPromise)
    ]

    private weak var controller: AppViewController?

    override public init() {
        super.init()
    }

    init(controller: AppViewController) {
        self.controller = controller
        super.init()
    }

    @objc func beginRefresh(_ call: CAPPluginCall) {
        controller?.beginPullToRefresh()
        call.resolve()
    }

    @objc func endRefresh(_ call: CAPPluginCall) {
        controller?.endPullToRefresh()
        call.resolve()
    }

    func notifyRefreshRequested() {
        notifyListeners("refreshRequested", data: [:], retainUntilConsumed: true)
    }
}

File: android/app/src/main/java/com/pessoauto/app/PullToRefreshPlugin.java
package com.pessoauto.app;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "PullToRefresh")
public class PullToRefreshPlugin extends Plugin {
    @PluginMethod
    public void endRefresh(PluginCall call) {
        if (getActivity() instanceof MainActivity) {
            ((MainActivity) getActivity()).endPullToRefresh();
        }
        call.resolve();
    }

    public void notifyRefreshRequested() {
        notifyListeners("refreshRequested", new JSObject(), true);
    }
}
